soql-standalone-parser/src/main/scala/com/socrata/NonEmptySeq.scala

```diff
@@ -3,28 +3,48 @@ package com.socrata
 // A collection that guarantees the existence of at least one element.
 // Fills the same role as scalaz's NonEmptyList, but now we don't have to include scalaz as a dependency,
 // and we can use Seq as the internal type (rather than List)
-case class NonEmptySeq[T](head: T, tail: Seq[T] = Nil) {
+case class NonEmptySeq[+T](head: T, tail: Seq[T] = Seq.empty) {
```

Comment:
It might be worth pulling some of these classes into a separate library some day. There's a good bit of code that isn't specific to SoQL that we might be able to utilize elsewhere.